encoding/rs/parametrized_encoder.go

```diff
@@ -0,0 +1,125 @@
+package rs
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/fft"
+	rb "github.com/Layr-Labs/eigenda/encoding/utils/reverseBits"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+)
+
+type ParametrizedEncoder struct {
```

Comment:
Refactor to make the encoder parametrized based on FFT settings (similar to Prover). Don't think it's too valuable since max time is ~600ms for loading largest fft setting. Am thinking of reverting